src/lib/logical_query_plan/alias_node.cpp

```diff
@@ -0,0 +1,44 @@
+#include "alias_node.hpp"
+
+#include <sstream>
+
+#include "boost/algorithm/string/join.hpp"
+#include "expression/expression_utils.hpp"
+
+namespace opossum {
+
+AliasNode::AliasNode(const std::vector<std::shared_ptr<AbstractExpression>> &expressions, const std::vector<std::string> &aliases):
```

Comment:
hmmm, not yet sure how I feel about having a node that only renames columns

Reply:
Thought about this back and forth for ages, really. In the end I concluded
* the aliases shouldn't be stored in the expression as they previously were. This would mess up hashes, comparing.
* there should be one thing that assigns aliases (`Alias{Node}`) and not two (`Projection{Node}, `Aggregate{Node}`)

I'll think about it again, but really, I feel the AliasNode/AliasOperator are the way to assign aliases.

Reply:
Maybe not a strong argument, but an own class/node for the aliases keeps the whole lqp/node concept consistent and easier to follow